recipes/gdcm/all/conanfile.py

```diff
@@ -0,0 +1,228 @@
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def validate(self):
+        if (self.settings.compiler == "Visual Studio"
+                and self.options.shared
+                and str(self.settings.compiler.runtime).startswith("MT")):
+            raise ConanInvalidConfiguration("shared gdcm can't be built with MT or MTd")
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, "11")
+
+    def requirements(self):
+        self.requires("zlib/1.2.11")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["GDCM_BUILD_DOCBOOK_MANPAGES"] = "OFF"
+        self._cmake.definitions["GDCM_BUILD_SHARED_LIBS"] = "ON" if self.options.shared else "OFF"
+        self._cmake.definitions["GDCM_USE_SYSTEM_ZLIB"] = "ON"
+
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("Copyright.txt", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        if self.settings.os == "Windows":
+            bin_dir = os.path.join(self.package_folder, "bin")
+            tools.remove_files_by_mask(bin_dir, "[!gs]*.dll")
+            tools.remove_files_by_mask(bin_dir, "*.pdb")
+        lib_dir = os.path.join(self.package_folder, "lib")
+        tools.rmdir(os.path.join(lib_dir, "gdcmopenjpeg-2.3"))
+        tools.rmdir(os.path.join(lib_dir, "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.remove_files_by_mask(os.path.join(lib_dir, self._gdcm_subdir), "[!U]*.cmake") #leave UseGDCM.cmake untouched
+        self._create_cmake_module_alias_targets(
+            os.path.join(self.package_folder, self._gdcm_cmake_module_aliases_path),
+            self._gdcm_libraries
+        )
+        self._create_cmake_variables(os.path.join(self.package_folder, self._gdcm_cmake_variables_path))
+
+    def _create_cmake_variables(self, variables_file):
+        v = tools.Version(self.version)
+        build_shared_libs = "ON" if self.options.shared else "OFF"
+        content = textwrap.dedent(f"""\
```

Comment:
we still support older python versions so please use `"".format()`

Reply:
repeats below =)

Reply:
fixed